src/battle/overlay_12_0224E4FC.c

```diff
@@ -6479,3 +6761,588 @@ BOOL CheckStatusEffectsSubstitute(BATTLECONTEXT *ctx, int battlerId, int status)
+        break;
+    case 1:
+        ctx->fieldSideConditionData[BattleSystem_GetFieldSide(bsys, battlerId)].mistTurns = data;
+        break;
+    case 2:
+        ctx->selectedMonIndex[battlerId] = data;
+        break;
+    case 3:
+        ctx->totalTurns = data;
+        break;
+    case 9:
+        ctx->unk_334.unk9C = data;
+        break;
+    case 11:
+        ctx->unk_334.unkA4[battlerId] = data;
+        break;
+    default:
+        GF_ASSERT(FALSE);
+    }
+}
+
+static const int ov12_0226CDCC[145] = {
+    0,
+	18, 
+    22, 
+    25, 
+    27,
+	31, 
+    47, 
+    37, 
+    14,
+	55, 
+    56, 
+    48, 
+    13,
+	58, 
+    63, 
+    12, 
+    12,
+	12, 
+    12, 
+    12, 
+    12,
+	12, 
+    12, 
+    12, 
+    12,
+	12, 
+    12, 
+    12, 
+    12,
+	64, 
+    66, 
+    85, 
+    86,
+	93, 
+    119, 
+    115, 
+    130,
+	138, 
+    147, 
+    12, 
+    12,
+	12, 
+    12, 
+    12, 
+    12,
+	12, 
+    12, 
+    12, 
+    12,
+	12, 
+    12, 
+    12, 
+    12,
+	44, 
+    142, 
+    149, 
+    150,
+	148, 
+    151, 
+    152, 
+    24,
+	33, 
+    34, 
+    35, 
+    43,
+	45, 
+    46, 
+    49, 
+    52,
+	54, 
+    62, 
+    67, 
+    68,
+	70, 
+    73, 
+    77, 
+    78,
+	79, 
+    80, 
+    81, 
+    82,
+	84, 
+    87, 
+    88, 
+    89,
+	91, 
+    92, 
+    95, 
+    96,
+	97, 
+    126, 
+    100, 
+    101,
+	103, 
+    105, 
+    106, 
+    109,
+	112, 
+    113, 
+    114, 
+    120,
+	122, 
+    123, 
+    124, 
+    125,
+	127, 
+    128, 
+    129, 
+    131,
+	132, 
+    134, 
+    135, 
+    140,
+	141, 
+    143, 
+    145, 
+    154,
+	155, 
+    156, 
+    158, 
+    159,
+	160, 
+    161, 
+    162, 
+    163,
+	164, 
+    165, 
+    166, 
+    167,
+	168, 
+    170, 
+    171, 
+    173,
+	175, 
+    218, 
+    219, 
+    220,
+	226, 
+    246, 
+    247, 
+    248,
+	249, 
+    260, 
+    261, 
+    118
+};
+
+static int ov12_02258348(BATTLECONTEXT *ctx, int statChangeType, u32 flag) {
+    ctx->statChangeType = statChangeType;
+    ctx->statChangeParam = flag & (0x7FFFFF);
+    ctx->statChangeFlag = flag & (0xFF800000);
+    
+    if (flag & (1 << 30)) {
+        ctx->battlerIdStatChange = ctx->battlerIdAttacker;
+    } else if (flag & (1 << 31)) {
+        ctx->battlerIdStatChange = ctx->battlerIdTarget;
+    } else if ((flag & (1 << 29)) || (flag & (1 << 28))) {
+        ctx->battlerIdStatChange = BATTLER_PLAYER;
+    }
+    
+    GF_ASSERT(NELEMS(ov12_0226CDCC) > (flag & (0x7FFFFF)));
+    
+    return ov12_0226CDCC[flag & (0x7FFFFF)];
+}
+
+static int ov12_022583B4(BATTLECONTEXT *ctx, int battlerId, int typeEffectiveness, int damage, int moveDamage, u32 *flag) {
+    if (!(ctx->battleStatus & BATTLE_STATUS_IGNORE_TYPE_EFFECTIVENESS) && !(ctx->battleStatus & BATTLE_STATUS_IGNORE_TYPE_IMMUNITY) && damage) {
+        damage = DamageDivide(damage * typeEffectiveness, 10);
+    }
+    
+    switch (typeEffectiveness) {
+    case TYPE_MUL_NO_EFFECT:
```

Comment:
the cases really need to be intended tbh, if not I'm likely just gonna go through and indent them anyway myself

Reply:
pls don't it starts looking really bad since GF likes to use triple embedded switch statements

Reply:
I mean diamond is already intended that way, the style guide is, and I have been indenting it for my cleanup PRs... which you have approved....